be/src/env/env_hdfs.cpp

```diff
@@ -50,19 +51,28 @@ static Status read_at_internal(hdfsFS fs, hdfsFile file, const std::string& file
                                                        offset, get_hdfs_err_msg()));
         }
     }
-    size_t bytes_read = 0;
-    while (bytes_read < res->size) {
-        size_t to_read = res->size - bytes_read;
-        auto hdfs_res = hdfsRead(fs, file, res->data + bytes_read, to_read);
-        if (hdfs_res < 0) {
+    if (starrocks::config::use_hdfs_pread || is_object_storage_path(file_name.c_str())) {
```

Comment:
This will cause string compare for each read. Can you eliminate this compare?

Reply:
done, PTAL